src/System.Net.WebSockets.Client/tests/CancelTest.cs

```diff
@@ -117,23 +117,52 @@ public class CancelTest : ClientWebSocketTestBase
                 await cws.CloseOutputAsync(WebSocketCloseStatus.NormalClosure, "CancelShutdown", cts.Token);
             }, server);
         }
+        
+        [OuterLoop]
+        [ConditionalTheory(nameof(WebSocketsSupported)), MemberData(nameof(EchoServers))]
+        public async Task ReceiveAsync_CancelThenReceive_ThrowsOperationCanceledException(Uri server)
```

Comment:
Are these tests running against .Net Framework as well? 
If not, could you copy/paste into a .Net Framework XUnit project and ensure that the behavior is the same?